app/src/test/java/com/nervousfish/nervousfish/service_locator/ServiceLocatorTest.java

```diff
@@ -0,0 +1,65 @@
+package com.nervousfish.nervousfish.service_locator;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public final class ServiceLocatorTest {
+    private IServiceLocator serviceLocator;
+
+    @Before
+    public void setup() {
+        serviceLocator = new ServiceLocator("foo");
+    }
+    @Test
+    public void testGetAndroidFilesDir() {
+        assertTrue("foo".equals(serviceLocator.getAndroidFilesDir()));
+    }
+
+    @Test
+    public void testGetDatabase() {
+        assertNotNull(serviceLocator.getDatabase());
```

Comment:
Actually, I don't think I can. I don't care what the service locator chooses as implementation for the interface, I just want to be sure that there is an implementation instead of null. Of course I can test if a specific instance is returned, but that's not the point of tests at all! Every module, the service locator, should be a black box so the tests should only test the behaviour of this black box, not the very exact classes that are used in the implementation. Also, who watches the watchers?